CMakeLists.txt

```diff
@@ -84,6 +84,17 @@ set_target_properties(relive_api      PROPERTIES UNITY_BUILD OFF) # too big
 set_target_properties(relive_api_test PROPERTIES UNITY_BUILD OFF) # too big
 # -----------------------------------------------------------------------------
 
+# -----------------------------------------------------------------------------
+# Setup GCC and Clang warnings
+add_library(project_warnings INTERFACE)
+export(TARGETS project_warnings FILE project_warnings.cmake)
+
+if(NOT MSVC)

```

Comment:
would we not want to kill the manual MSVC warning stuff I added and use this instead?